Recipe book

Emdorkable opened the pull request:
new firebase, recipe book

danthynguyen committed:
Merge branch 'main' of https://github.com/jcderry5/hci-recipe-app

Emdorkable committed:
what i have so far for main menu

Emdorkable committed:
Merge branch 'main' of https://github.com/jcderry5/hci-recipe-app into profile

danthynguyen committed:
Merge branch 'main' of https://github.com/jcderry5/hci-recipe-app

Emdorkable committed:
working on format

Emdorkable committed:
Merge branch 'main' of https://github.com/jcderry5/hci-recipe-app into profile

danthynguyen committed:
Merge branch 'main' of https://github.com/jcderry5/hci-recipe-app

Emdorkable committed:
old firebase recipe-book

Emdorkable committed:
old firebase recipe-book wait

Emdorkable committed:
new firebase recipebook

willeng37 committed:
Fix null user error via console.log print

willeng37 commented:
I think the only issues that probably need to be checked are rendering errors due to a null (non-logged in) user for any protected pages.

danthynguyen reviewed (dismissed):
the recipes are not rendering on the recipe book page

there is an uncaught promise in RecipeBook.js 44:1, looks like its trying to convert an undefined or null into an object

![image](https://user-images.githubusercontent.com/73391187/230225032-606d96ec-a170-4078-aebb-88515a3d6570.png)

I think there might be something wrong with the fetch in the function

danthynguyen reviewed (approved):
list of issues to address later
- only renders on refresh on first time
- when reaccessing the recipe book, takes 5 seconds to render, otherwise looking at a blank page
- memory-wasting on firebase

danthynguyen committed:
Merge branch 'main' into recipe-book

danthynguyen committed:
merge

jcderry5 commented on src/components/NewRecipe.js:
```diff
@@ -137,8 +142,9 @@ export default function NewRecipe() {
-                recipeIndex={currentRecipeIndex}
-                currentIngredients={currentIngredients} />
+            currentIngredients={currentIngredients} 
+            changeAddedIngredients={changeAddedIngredients}
```
remove changeAddedIngredients :)

jcderry5 committed:
Cleaning and formatting code

jcderry5 reviewed (approved):
LGTM

willeng37 committed:
Render firebase database on load once.

Utilizes useEffect() and user state to only render when user's authentication has been confirmed.

Should remove any looping calls to Firebase.

jcderry5 reviewed (approved):
Trusting Will
